src/teacher.py

```diff
@@ -72,3 +74,32 @@ def _answer_Demonstration_query(
                                           _is_demo=True,
                                           _train_task_idx=query.train_task_idx)
         return DemonstrationResponse(query, teacher_traj)
+
+
+@dataclass
+class TeacherInteractionMonitor(Monitor):
+    """Wraps the interaction between agent and teacher."""
+    _request: InteractionRequest
+    _teacher: Teacher
+    _responses: List[Optional[Response]] = field(init=False)
+    _query_cost: float = field(init=False, default=0.0)
+
+    def __post_init__(self) -> None:
+        self._responses = []
```

Comment:
instead, use the pattern in the question here:
https://stackoverflow.com/questions/53632152/why-cant-dataclasses-have-mutable-defaults-in-their-class-attributes-declaratio

while we can't do:
_responses: List[Optional[Response]] = field(init=False, default=[])
we can do:
_responses: List[Optional[Response]] = field(init=False, default_factory=list)

Reply:
coolio